tools/boundary/glorys_obc_workflow/mom6_obc_workflow.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+set -eu
+
+# Load required modules and environments
+source $MODULESHOME/init/sh
+module load miniforge
+conda activate /nbhome/role.medgrp/.conda/envs/uwtools || { echo "Error activating conda environment. Exiting."; exit 1; }
+
+# Helper functions
+print_usage() {
+    echo "Usage: $0 START_DATE END_DATE [--ncrcat] [--adjust-timestamps]"
+    echo "  START_DATE and END_DATE must be in YYYY-MM-DD format."
+    echo "  --ncrcat: Enable ncrcat step (skips subset, fill, and submit_python steps)."
+    echo "  --adjust-timestamps: Adjust timestamps during ncrcat step."
+}
+
+validate_date_format() {
+    if [[ ! "$1" =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ ]]; then
+        echo "Error: Date $1 must be in YYYY-MM-DD format."
+        exit 1
+    fi
+}
+
+log_message() {
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] $1"
+}
+
+# Default options
+DO_NCRCAT=false
+ADJUST_TIMESTAMPS=false
+PYTHON_SCRIPT="../write_glorys_boundary_daily.py"
+
+# Parse arguments
+START_DATE="$1"
+END_DATE="$2"
+shift 2
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --ncrcat)
+            DO_NCRCAT=true
+            ;;
+        --adjust-timestamps)
+            ADJUST_TIMESTAMPS=true
+            ;;
+        *)
+            echo "Unknown argument: $1"
+            print_usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+validate_date_format "$START_DATE"
+validate_date_format "$END_DATE"
+
+
+start_date_epoch=$(date -d "$START_DATE" +%s)
+end_date_epoch=$(date -d "$END_DATE" +%s)
+if [[ $start_date_epoch -gt $end_date_epoch ]]; then
+    log_message "Error: START_DATE ($START_DATE) must not be after END_DATE ($END_DATE). Exiting."
+    exit 1
+fi
+
+# Ensure --adjust-timestamps is only used with --ncrcat
+if $ADJUST_TIMESTAMPS && ! $DO_NCRCAT; then
+    echo "Error: --adjust-timestamps can only be used with --ncrcat."
+    exit 1
+fi
+
+# Warn user when --ncrcat is enabled
+if $DO_NCRCAT; then
+    log_message "WARNING: --ncrcat is enabled. The script will SKIP subset, fill, and submit_python steps."
+    log_message "Ensure that all daily outputs already exist for the specified date range."
+fi
+
+# Prepare directories
+CURRENT_DATE=$(date +%Y-%m-%d-%H-%M)
+mkdir -p ./log/$CURRENT_DATE ./outputs scripts
+
+# Define user configurations
```

Comment:
maybe change the default emails and paths in the section to use  `$USER`, so if users forget to change one or two lines, you don't get flooded with a bunch of emails!